src/plugins/ui_feature/selection_input.ts

```diff
@@ -181,13 +182,14 @@ export class SelectionInputPlugin extends UIPlugin implements StreamCallbacks<Se
    * Insert new inputs after the given index.
    */
   private insertNewRange(index: number, values: string[]) {
+    const currentMaxId = Math.max(0, ...this.ranges.map((range) => Number(range.id)));
     this.ranges.splice(
       index,
       0,
       ...values.map((xc, i) => ({
         xc,
-        id: (this.ranges.length + i + 1).toString(),
-        color: colors[(this.ranges.length + i) % colors.length],
+        id: (currentMaxId + i + 1).toString(),
```

Comment:
you start at index 1, which means you skip `colors[0]`, why was this change required ? I don't see any test related ?

Reply:
the typing of id is UID, which means it could pretty much be any kind of string - I think it'd be way better to drop the UID and work with numbers don't you think ?

Reply:
@rrahir I changed it because in the old implementation `id` can be duplicate in some cases, which causes the some problems when focusing and updating input. My bad! I didn't notice the color will also start at index 1. I can change the calculation of `currentMaxId` to compare to `-1`. 
For the type it seems to be a legacy problem (?) But I agreed with you. Number type is better. I can change it in another PR.